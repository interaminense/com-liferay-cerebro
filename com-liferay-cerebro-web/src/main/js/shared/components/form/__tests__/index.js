import Form from '../index';

describe('Form', () => {
	let component;

	afterEach(() => {
		if (component) {
			component.dispose();
		}
	});

	it('should render a Form', () => {
		component = new Form();

		expect(component).toMatchSnapshot();
	});

	it('should add item to formElements array', () => {
		component = new Form();

		component.addFormElement(1);

		expect(component._formElements).toEqual([1]); // eslint-disable-line no-underscore-dangle
	});

	it('should remove item to formElements array', () => {
		component = new Form();

		component._formElements = [1]; // eslint-disable-line no-underscore-dangle

		component.removeFormElement(1);

		expect(component._formElements).toEqual([]); // eslint-disable-line no-underscore-dangle
	});

	it('should prevent Form from submitting if validate is false', () => {
		component = new Form();

		component.validate = jest.fn(() => false);

		const spy = jest.fn();

		component.handleSubmit({preventDefault: spy});

		expect(spy).toHaveBeenCalled();
	});

	it('should call onChange when a descendant calls informChange', () => {
		const onChange = jest.fn();

		component = new Form({
			onChange
		});

		expect(onChange).not.toHaveBeenCalled();

		component.getChildContext().ancestorForm.informChange('foo');

		expect(onChange).toHaveBeenCalledWith({}, 'foo');
	});

	it('should call onSubmit when the form is submitted', () => {
		const onSubmit = jest.fn();

		component = new Form({
			onSubmit
		});

		expect(onSubmit).not.toHaveBeenCalled();

		component.handleSubmit({preventDefault: jest.fn()});

		expect(onSubmit).toHaveBeenCalledWith({});
	});

	it('should render a Form Group', () => {
		component = new Form.Group();

		expect(component).toMatchSnapshot();
	});

	it('should render an inline Form Group', () => {
		component = new Form.Group({inline: true});

		expect(component).toMatchSnapshot();
	});

	it('should render an autofit Form Group', () => {
		component = new Form.Group({autoFit: true});

		expect(component).toMatchSnapshot();
	});
});