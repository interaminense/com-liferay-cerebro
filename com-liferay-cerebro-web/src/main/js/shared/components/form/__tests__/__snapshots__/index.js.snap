// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should render a Form 1`] = `<form class="form-root"></form>`;

exports[`Form should render a Form Group 1`] = `<div class="form-group"></div>`;

exports[`Form should render an autofit Form Group 1`] = `<div class="form-group form-inline-autofit"></div>`;

exports[`Form should render an inline Form Group 1`] = `<div class="form-group form-inline"></div>`;
